controllers/construct/mongodbstatefulset.go

```diff
@@ -88,7 +86,7 @@ type MongoDBStatefulSetOwner interface {
 	LogsVolumeName() string
 
 	// GetMongodConfiguration returns the MongoDB configuration for each member.
-	GetMongodConfiguration() map[string]interface{}
+	GetMongodConfiguration() mdbv1.MongodConfiguration
```

Comment:
this interface is used in enterprise (for deploying the AppDB). I think this will still be fine, but next time we update the community operator version library in the enterprise repo we can address this.

Reply:
It might also be a good reason to synchronise the way we're handling the `mongod` options in the two operators.